checker/src/main/java/org/checkerframework/checker/nullness/CollectionToArrayHeuristics.java

```diff
@@ -35,26 +43,48 @@
  *       </ol>
  * </ol>
  *
- * Note: The nullness of the returned array doesn't depend on the passed array nullness.
+ * <p>Additionally, when the lint option {@link NullnessChecker#LINT_TRUSTARRAYLENZERO} is provided,
+ * field accesses where the field declaration has a {@code @ArrayLen(0)} annotation are considered
+ * when determining whether a call to {@link Collection#toArray(Object[]) Collection.toArray(T[])}
+ * will return an array with non-null components. This trusts the {@code @ArrayLen(0)} annotation,
+ * but does not verify it. Run the Constant Value Checker to verify that annotation.
+ *
+ * <p>Note: The nullness of the returned array doesn't depend on the passed array nullness.
```

Comment:
Is this a fact about nullness, or is it a fact about the heuristics defined in this routine?  Please clarify.

Reply:
This is a fact about `toArray`. It's from the original description.

Reply:
Could you clarify it?  If it's unclear wherever the original description is, then please clarify that too.  (And please link to it.)